src/services/addTokenToExchange.ts

```diff
@@ -0,0 +1,53 @@
+import { ExchangeApi, Erc20Api } from 'types'
+import { getErc20Info } from './getErc20Info'
```

Comment:
I think we should make a rule on never using other services to implement new services. We can easily get into cyclic dependency problems.

One solution is, to have reusable helper functions to refactor this pieces of code. I.e. Here you could make `./_getErc20Info` a function that requires all APIs+params. The service layers use the helper providing both things

Reply:
Agree. Will play around with what might be best. Initial idea: split code into sub dirs:

```
services/
  index.ts
  factories/
    getTokenFromExchange.ts
    addTokenToExchange.ts
  helpers/
    getErc20Info.ts
```